### DISCUSSION TITLE
Automatically sync store to DB

### DISCUSSION THREAD

**Discussion by unknown1337** [CodeDredd/pinia-orm#206] [Q&A] [answered]
Dear,

I want to provide functionality to the pinia repositories to automatically sync store changes to the database. 

I created a working new method (e.g. destroySync) on a **specific** modelRepo that: 
a. performs the original pinia ORM destroy functionality 
b. AND performs the same action on a connected db 
c. OnDbError add the removed entries back to the store (to be in sync with the db).

```
import { Repository } from "pinia-orm"; import Workspace from "./workspaceModel"; import { api } from "src/services/apiService"; import { 
 supabase } from "src/boot/auth"; import { remove } from 'lodash'; import { log } from "src/boot/log"; 

export class WorkspaceRepo extends Repository {
  use = Workspace

  async destroySync(ids: string[]) {
    log.debug(`Destroy sync called for ids: ${ids}`)
    const entity = this.model.$entity()
    const objs = this.query().whereIn('id', ids).get()

    this.destroy(ids)
    const deletedItems = await api.process(() => supabase.from(entity).delete().filter('id', 'in', `(${ids})`))

    // remove deletedItems from objs array
    deletedItems?.map((deletedItem) => { remove(objs, (I) => I.id === deletedItem.id) })

    // not all were deleted, restore the ones that were not deleted
    if (objs) this.save(objs);

  }

}
```
Question: 
1. How can I add this function to **EVERY** pinia model/ (store/) repository?
2) is there a more clever way to achieve the desired result? thanks!**

MANY THANKS

_Note: I'm using supabase client js,  quasar V2, Vue V3_, pinia persistance, pinia ORM_

**Comment by CodeDredd** [comment, marked as answer]
For now what comes to mind is making an own `CustomRepository` class extending `Repository` which has this methods for all your repositories.

And some point i will make the plugin system work and then it should be possible to define it in the `createORM` process

**Reply by unknown1337** [reply]
dear @CodeDredd , thanks for your reply! 

1) I've already got custom repositorie**sssss** for all my models :). 
2) each custom repository uses a **single** model: use = Model
3) The thing I'm looking for is to define a function that will automatically be part of every custom repository (without copy & paste work to 10 repositories:P).  
4) how can I achieve point 3? Can I create a custom repo that extends all models/..? or is there a better way? 

Thanks!

**Reply by CodeDredd** [reply]
@unknown1337 ok i was a bit unclear
I ment that you should crate a `BaseRepository` extending `Repository` which has no `use` and then you all your `CustomRepositories` are extending `BaseRepository`. Hopefully its now clear 😄

**Reply by unknown1337** [reply]
Clear, thanks!
